fsm.go

```diff
@@ -76,36 +76,40 @@ func (r *fsmS) scheduleRetryWithExponentialDelay(e *f.Event, cb func(e *f.Event)
 
 func (r *fsmS) lookupAgentHost(e *f.Event) {
 	cb := func(found bool, host string) {
+		// Agent host is found through the checkHost method, that attempts to read "Instana Agent" from the response header.
 		if found {
 			r.lookupSuccess(host)
 			return
 		}
 
-		gateway, err := getDefaultGateway("/proc/net/route")
-		if err != nil {
-			r.agent.logger.Error("failed to fetch the default gateway, scheduling retry: ", err)
-			r.scheduleRetry(e, r.lookupAgentHost)
+		if runtime.GOOS == "Linux" {
```

Comment:
```suggestion
		if _, err := os.Stat("/proc/net/route"); err == nil {
```

What do you think about this? Shall we check if the file exists, instead of if OS is Linux?